server/src/main/java/org/opensearch/common/xcontent/support/XContentHttpChunk.java

```diff
@@ -15,10 +15,15 @@
 import org.opensearch.core.xcontent.XContentBuilder;
 import org.opensearch.http.HttpChunk;
 
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+
 /**
  * Wraps the instance of the {@link XContentBuilder} into {@link HttpChunk}
  */
 public final class XContentHttpChunk implements HttpChunk {
+    private static final byte[] CHUNK_SEPARATOR = new byte[] { '\r', '\n' };
```

Comment:
Probably not super important, but is there a reason to prefer `\r\n` over the single byte `\n`?

Reply:
Yes, for example, the (single) chunk in response may looks like this: 
```
{ "index": { "_index": "test-streaming", "_id": "3" } }
{ "name": "john" }
```

(we don't emit those but it is possible to have multi-line here)

Reply:
Ooooh, so lines within a chunk will be separated by `\n` and chunks are separated by `\r\n`?

Reply:
If that's the case, my instinct is that given that newline delimited JSON is human readable and that `\n` and `\r\n` are rendered the same, maybe we should use a printable character as part of the chunk delimiter?

Reply:
> If that's the case, my instinct is that given that newline delimited JSON is human readable and that `\n` and `\r\n` are rendered the same, maybe we should use a printable character as part of the chunk delimiter?

Chunks are separated by `\r\n` (see please https://en.wikipedia.org/wiki/Chunked_transfer_encoding), it just we have to kind of keep the separator within chunk itself.